lib/finch/pool.ex

```diff
@@ -55,42 +61,42 @@ defmodule Finch.Pool do
   end
 
   @impl NimblePool
-  def init_pool({registry, shp, opts}) do
-    {:ok, _} = Registry.register(registry, shp, [])
-    {:ok, {shp, opts}}
+  def init_pool({registry, state}) do
+    {:ok, _} = Registry.register(registry, state.shp, state.registry_value)
```

Comment:
I realize that we will have this registry value duplicated for every single pool, but I think this is the best way to quickly look it up at the same time as the pool pid.